crates/sui-types/src/base_types.rs

```diff
@@ -673,29 +660,11 @@ impl From<&MultiSigPublicKey> for SuiAddress {
 }
 
 /// Sui address for [struct ZkLoginAuthenticator] is defined as the black2b hash of
-/// [zklogin_flag || iss_bytes_length || iss_bytes || address_seed in bytes].
+/// [zklogin_flag || iss_bytes_length || iss_bytes || unpadded_address_seed_in_bytes].
 impl TryFrom<&ZkLoginAuthenticator> for SuiAddress {
     type Error = SuiError;
     fn try_from(authenticator: &ZkLoginAuthenticator) -> SuiResult<Self> {
-        TryFrom::try_from(&authenticator.inputs)
-    }
-}
-
-/// Sui address for [struct ZkLoginAuthenticator] is defined as the black2b hash of
-/// [zklogin_flag || iss_bytes_length || iss_bytes || address_seed in bytes].
-impl TryFrom<&ZkLoginInputs> for SuiAddress {
```

Comment:
was this code redundant? (we don't move it)

Reply:
this is now calls the explicit definition in SuiAddress instead of re-implementing it:

```
    /// Define as iss_bytes_len || iss_bytes || unpadded_32_byte_address_seed.
    pub fn try_from_unpadded(inputs: &ZkLoginInputs) -> SuiResult<Self> {
        let mut hasher = DefaultHash::default();
        hasher.update([SignatureScheme::ZkLoginAuthenticator.flag()]);
        let iss_bytes = inputs.get_iss().as_bytes();
        hasher.update([iss_bytes.len() as u8]);
        hasher.update(iss_bytes);
        // this converts an address seed from bigint to bytes, length can be shorter than 32 bytes and left unpadded.
        hasher.update(
            big_int_str_to_bytes(inputs.get_address_seed())
                .map_err(|_| SuiError::InvalidAddress)?,
        );
        Ok(SuiAddress(hasher.finalize().digest))
    }```